evap/student/templates/student_index.html

```diff
@@ -51,8 +51,7 @@
                             {% for course in semester.courses %}
                                 <tr{% if course.state == 'in_evaluation' and course not in voted_courses %}
                                         class="hover-vote" data-url="{% url 'student:vote' course.id %}"
-                                    {# make results row hoverable only if course is published so reviewers cannot click on it before #}
-                                    {% elif course.state == 'published' and course|can_user_see_results_page:request.user %}
+                                    {% elif course.state == 'published' and course.can_publish_rating_results %}
```

Comment:
the problem was (and is), that staff users have other view permissions for the results page but shouldn't be able to access the results page from the student's index if there are no results. because in this situation they are using the page as students (same as with the public/own view on the results pages).
this solution now checks for what is relevant for students: they can only access the page if the results were published. this makes the page behave correctly even for staff users.

Reply:
right. given my repeated confusion, i'd still like to have a comment there. maybe "staff users should be able to access courses through the student index only if it is actually has published results"?